prog/dftb+/lib_dftbplus/main.F90

```diff
@@ -1361,6 +1505,9 @@ subroutine reallocateSparseArrays(sparseSize, ham, over, H0, rhoPrim, iHam, iRho
       deallocate(ERhoPrim)
       allocate(ERhoPrim(sparseSize))
     end if
+    if (allocated(reks)) then
+      call REKS_reallocate(reks, sparseSize)
```

Comment:
Please implement it as type bound procedure, so that the call looks as `call reks%reallocate(sparseSize)`